apps/frontend/src/components/DummyData/DummyData.tsx

```diff
@@ -0,0 +1,68 @@
+'use client';
+import { useInfiniteQuery } from '@tanstack/react-query';
+import { dummyDataApi } from 'api-client';
+import { useEffect, useState } from 'react';
+
+import useScrollbar from 'src/hooks/useScrollbar';
+
+import { BrowseProfileCard } from '../cards/BrowseProfileCard';
+
+import type { DummyDataProps } from './types';
+
+export default function DummyData({ initialData }: DummyDataProps) {
+  const limit = 10;
+  //shifted because initial data is first 10 profiles
+  const [offset, setOffset] = useState(10);
+
+  const fetchProfiles = async () => {
+    const { data } = await dummyDataApi.getData({ limit, offset }).throwOnError();
+    setOffset((x) => x + limit);
+
+    return data;
+  };
+
+  const { data, error, fetchNextPage, hasNextPage, isFetchingNextPage } = useInfiniteQuery({
+    queryKey: ['profiles'],
+    queryFn: fetchProfiles,
+    staleTime: Infinity, //to prevent refetch initial data
+    initialPageParam: 0,
+    initialData: () => {
+      return { pageParams: [null], pages: [initialData] };
+    },
+    getNextPageParam: (lastPage, allPages) =>
+      lastPage.length === limit ? allPages.length * limit : undefined,
+  });
+
+  if (error) {
+    throw error;
```

Comment:
Do not rethrow error if it happened on the client side. You can just show a toast.